static/js/keydown_util.ts

```diff
@@ -0,0 +1,58 @@
+import type $ from "jquery";
+
+/*
+    See hotkey.js for handlers that are more app-wide.
+*/
+
+// Note that these keycodes differ from those in hotkey.js, because
+// we're using keydown rather than keypress.  It's unclear whether
+// there's a good reason for this difference.
+const keys = {
+    Enter: "enter_key", // Enter
+    ArrowLeft: "left_arrow", // // Left arrow
+    ArrowUp: "up_arrow", // Up arrow
+    ArrowRight: "right_arrow", // Right arrow
+    ArrowDown: "down_arrow", // Down arrow
+    h: "vim_left", // 'H'
+    j: "vim_down", // 'J'
+    k: "vim_up", // 'K'
+    l: "vim_right", // 'L'
+} as const;
+
+type Keys = typeof keys;
+
+export function handle(opts: {
+    elem: ReturnType<typeof $>;
```

Comment:
```suggestion
    elem: JQuery;
```

This type is global and doesn’t need to be imported, but it does need an `.eslintrc.json` exemption. (I’m not totally happy about it being global, but I’m okay with this exemption because TypeScript should catch any incorrect attempt to use `JQuery` as a value rather than a type.)

```diff
@@ -124,12 +124,15 @@
         {
             "files": ["**/*.ts"],
             "extends": ["plugin:@typescript-eslint/recommended", "plugin:import/typescript"],
             "parserOptions": {
                 "project": "tsconfig.json"
             },
+            "globals": {
+                "JQuery": false
+            },
             "rules": {
                 // Disable base rule to avoid conflict
                 "no-duplicate-imports": "off",
                 "no-unused-vars": "off",
                 "no-useless-constructor": "off",
 
```